src/message.rs

```diff
@@ -40,14 +48,19 @@ impl Decoder for Codec {
 
         let varint_len = src.len() - rest.len();
 
-        // TODO: if len > MAX return error
+        if varint_len > MAX_MESSAGE_SIZE {
+            return Err(io::Error::new(io::ErrorKind::Other, "Message too large"));
```

Comment:
```suggestion
            return Err(io::Error::other("Message too large"));
```

Reply:
This will increase MSRV to 1.74.0. Now we are at 1.73.0. What do you think?

Reply:
I thought we don't care :sweat_smile:

Reply:
applied the patch